### PR TITLE
Permission editor main page

### PR DESCRIPTION
**Pull request by pusztaienike**

### CODE REVIEW

File: apps/sensenet/src/components/view-controls/permission-view.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                      }
+                    />
+                  </ListItem>
+                )
+              })}
+          </List>
+        </Collapse>
+        <ListItem button onClick={() => setOpenSetOnThisList(!openSetOnThisList)}>
+          {openSetOnThisList ? <ExpandLess /> : <ExpandMore />}
+          <ListItemText primary="Set on this content" className={classes.listTitle} />
```
Comment (herflis) [inline apps/sensenet/src/components/view-controls/permission-view.tsx:180]: Could this primary text be localized?

File: apps/sensenet/src/components/view-controls/permission-view.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+          <span className={classes.textBolder}>{currentContent?.DisplayName}</span>
+        </div>
+        <Button aria-label={localization.permissionEditor.assign} color="primary" variant="contained">
+          {localization.permissionEditor.assign}
+        </Button>
+      </div>
+      <List component="nav">
+        <ListItem button onClick={() => setOpenInheritedList(!openInheritedList)}>
+          {openInheritedList ? <ExpandLess /> : <ExpandMore />}
+          <ListItemText primary="Inherited from ancestor" className={classes.listTitle} />
```
Comment (herflis) [inline apps/sensenet/src/components/view-controls/permission-view.tsx:129]: Could this primary text be localized? :)

File: apps/sensenet/src/components/view-controls/permission-view.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      })
+      setCurrentContent(result.d)
+    }
+    getCurrentContent()
+  }, [props.contentPath, repo])
+
+  useEffect(() => {
+    async function getAllPermissions() {
+      try {
+        const result = await repo.executeAction<any, AclResponseType>({
```
Comment (herflis) [inline apps/sensenet/src/components/view-controls/permission-view.tsx:97]: I think GetAcl should be moved to `repository.security` in the client-core package

**Review by herflis — APPROVED** [review]

File: apps/sensenet/src/components/view-controls/permission-view.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                )
+              })}
+          </List>
+        </Collapse>
+        <ListItem button onClick={() => setOpenSetOnThisList(!openSetOnThisList)}>
+          {openSetOnThisList ? <ExpandLess /> : <ExpandMore />}
+          <ListItemText primary={localization.permissionEditor.setOnThis} className={classes.listTitle} />
+        </ListItem>
+        <Collapse className={classes.collapseWrapper} in={openSetOnThisList} timeout="auto" unmountOnExit>
+          {permissions?.entries.filter((entry) => entry.inherited === false).length === 0 ? (
```
Comment (taki9) [inline apps/sensenet/src/components/view-controls/permission-view.tsx:179]: store the filtered array in a variable to gain performance

File: packages/sn-client-core/src/Models/SecurityModels.ts [hunk 5]
```diff
@@ -101,3 +101,52 @@ export interface PermissionResponseModel {
   inherits: boolean
   entries: PermissionEntry[]
 }
+
+export class PermissionType {
```
Comment (taki9) [inline packages/sn-client-core/src/Models/SecurityModels.ts:105]: use interfaces for types

File: packages/sn-client-core/src/Repository/Security.ts [hunk 6]
```diff
@@ -227,6 +227,20 @@ export class Security {
...
+   * Required permissions to call this action: See permissions.
+   * @param {string | number} contentIdOrPath The path or id for the content
+   * @returns {Promise<PermissionResponseModel>} A promise with the permission response
+   */
+  public getAcl = (contentIdOrPath: string | number) =>
+    this.repository.executeAction<any, AclResponseModel>({
+      idOrPath: contentIdOrPath,
+      name: 'GetAcl',
+      method: 'GET',
+      body: undefined,
```
Comment (taki9) [inline packages/sn-client-core/src/Repository/Security.ts:241]: get request has no body, so simply remove this option here

File: apps/sensenet/src/components/view-controls/permission-view.tsx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+      width: '40px',
+    },
+    collapseWrapper: {
+      paddingLeft: '30px',
+    },
+    anchor: {
+      fontSize: '14px',
+      color: theme.palette.primary.main,
+      paddingLeft: '15px',
+      cursor: 'pointer',
```
Comment (taki9) [inline apps/sensenet/src/components/view-controls/permission-view.tsx:59]: color, cursor and hover underline are already set by material-ui classes

**Review by taki9 — CHANGES_REQUESTED** [review]

**Comment by taki9** [comment]
I think a Cancel button would be useful in the right bottom corner.

**Comment by pusztaienike** [comment]
> I think a Cancel button would be useful in the right bottom corner.

Done

**Review by taki9 — APPROVED** [review]

**Pull request merged**